LibLancBlocking.cpp

```diff
@@ -0,0 +1,391 @@
+    {
+        return false;
+    };
+    startTime = waitForStartBit();
+    if (!receiveByte(&_transmitReceiveBuffer[4], startTime))
+    {
+        return false;
+    };
+    startTime = waitForStartBit();
+    if (!receiveByte(&_transmitReceiveBuffer[5], startTime))
+    {
+        return false;
+    };
+    startTime = waitForStartBit();
+    if (!receiveByte(&_transmitReceiveBuffer[6], startTime))
+    {
+        return false;
+    };
+    startTime = waitForStartBit();
+    if (!receiveByte(&_transmitReceiveBuffer[7], startTime))
+    {
+        return false;
+    };
+
+    return true; // loop completed without issue,  return true.
+}
+
+bool LancBlocking::transmitByte(uint8_t byte, unsigned long startTime)
+{
+    transmitZero();
+    for (uint8_t i = 0; i < 8; i++)
+    {
+        if (byte & (1 << i))
+        {
+            transmitOne();
+        }
+        else
+        {
+            transmitZero();
+        }
+        delayUsWithStartTime(startTime, (i + 1) * LANC_BIT_TIME_US); // Wait for the bit to be transmitted
+    }
+
+    // Check if the total time to send the data is within tollerance.  If not set lockout and return false.
+    //  This should not really ever occur unless microcontroller is running to slow.
+    unsigned long ttime = micros() - startTime;
+    if (ttime >= ((8 * LANC_BIT_TIME_US) + LANC_HALF_BIT_TIME_US))
+    {
+        LANC_FUNCTION_LOCKOUT = true;
+        ClearCommand();
+        return false;
+    }
+
+    transmitZero(); // Complete transmission
+    return true;
+}
+bool LancBlocking::receiveByte(uint8_t *byte, unsigned long startTime)
+{
+    *byte = 0;
+    for (uint8_t i = 0; i < 8; i++)
+    {
+        if (inputState())
+        {
+            *byte |= 1 << i;
+        }
+        delayUsWithStartTime(startTime, (i + 1) * LANC_BIT_TIME_US);
+    }
+
+    // Check if the total time to receive the data is within tollerance.  If not set lockout and return false.
+    //  This should not really ever occur unless microcontroller is running to slow.
+    unsigned long ttime = micros() - startTime;
+    if (ttime >= ((8 * LANC_BIT_TIME_US) + LANC_HALF_BIT_TIME_US))
+    {
+        LANC_FUNCTION_LOCKOUT = true;
+        ClearCommand();
+        return false;
+    }
+    return true;
+}
+
+void LancBlocking::transmitOne()
+{
+    digitalWrite(_outputPin, HIGH);
+}
+
+void LancBlocking::transmitZero()
+{
+    digitalWrite(_outputPin, LOW);
+}
+
+bool LancBlocking::inputState()
+{
+    return digitalRead(_inputPin);
+}
+
+// Sync to next LANC message
+// lanc protocol requires a stop condition that is longer than 5 milliseconds. Since we expect this function
+// to be called periodically and thus have a delay between completion and next start we use a lower value that
+// is still enough to determine between stop conditions during an ongoing transmission and the stop condition
+// between two transmissions.
+unsigned long LancBlocking::syncTransmission()
```

Comment:
Can you elaborate why you made those changes? This method was very small and easy to understand. Now it got very complicated. I think something is not quite right here.....